h2/src/test/org/h2/test/db/TestGeneralCommonTableQueries.java

```diff
@@ -469,4 +469,82 @@ private void testColumnNames() throws Exception {
         conn.close();
         deleteDb("commonTableExpressionQueries");
     }
+    
+    private void testRecursiveTable() throws Exception {
+        String[] expectedRowData =new String[]{"|meat|null","|fruit|3","|veg|2"};
+        String[] expectedColumnNames =new String[]{"VAL",
+                "SUM(SELECT     X FROM (     SELECT         SUM(1) AS X,         A     FROM PUBLIC.C     INNER JOIN PUBLIC.B         ON 1=1     WHERE B.VAL = C.B     GROUP BY A ) BB WHERE BB.A IS A.VAL)"};
+        
+        deleteDb("commonTableExpressionQueries");
+        Connection conn = getConnection("commonTableExpressionQueries");
+        PreparedStatement prep;
+        ResultSet rs;
+        
+        String SETUP_SQL = 
+             "DROP TABLE IF EXISTS A;                           "
+            +"DROP TABLE IF EXISTS B;                           "
+            +"DROP TABLE IF EXISTS C;                           "
+            +"CREATE TABLE A(VAL VARCHAR(255));                 "
+            +"CREATE TABLE B(A VARCHAR(255), VAL VARCHAR(255)); "
+            +"CREATE TABLE C(B VARCHAR(255), VAL VARCHAR(255)); "
+            +"                                                  "
+            +"INSERT INTO A VALUES('fruit');                    "
+            +"INSERT INTO B VALUES('fruit','apple');            "
+            +"INSERT INTO B VALUES('fruit','banana');           "
+            +"INSERT INTO C VALUES('apple', 'golden delicious');"
+            +"INSERT INTO C VALUES('apple', 'granny smith');    "
+            +"INSERT INTO C VALUES('apple', 'pippin');          "
+            +"INSERT INTO A VALUES('veg');                      "
+            +"INSERT INTO B VALUES('veg', 'carrot');            "
+            +"INSERT INTO C VALUES('carrot', 'nantes');         "
+            +"INSERT INTO C VALUES('carrot', 'imperator');      "
+            +"INSERT INTO C VALUES(null, 'banapple');           "
+            +"INSERT INTO A VALUES('meat');                     "
+            ;
+        String WITH_QUERY =
+            "WITH BB as (SELECT                        \n"
+            +"sum(1) as X,                             \n"
+            +"a                                        \n"
+            +"FROM B                                   \n"
+            +"JOIN C ON B.val=C.b                      \n"
+            +"GROUP BY a)                              \n"
+            +"SELECT                                   \n"
+            +"A.val,                                   \n"
+            +"sum(SELECT X FROM BB WHERE BB.a IS A.val)\n"//AS SUM_X
+            +"FROM A                                   \n"
+            +"GROUP BY A.val";
+
+        for(@SuppressWarnings("unused") int queryRunTries: new int[]{1,2,3}){
```

Comment:
this seems a little clever compared to a normal for loop :-)

Reply:
Reverted